src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final double MAX_SIZE = 10.0;
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final double DEFAULT_RADIUS = 10.0;
+
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = RANDOM.nextInt(5);
+
+        switch (figureType) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(),
+                        RANDOM.nextDouble() * MAX_SIZE);
+            case 1:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        RANDOM.nextDouble() * MAX_SIZE,
+                        RANDOM.nextDouble() * MAX_SIZE);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.